### PR TITLE
Checkpoint resolver functionality and improvements.

### PR DESCRIPTION
**Pull request by digimaun**
* Adds `ResolutionSettings` which can be optionally passed in (see readme).
* Supports resolution from pre-calculated deps (*.expanded.json)
* Supporting Readme updates
* Tests

### CODE REVIEW

File: resolvers/dotnet/Azure.DigitalTwins.Resolver/ModelQuery.cs [hunk 1]
```diff
@@ -100,6 +102,34 @@ public IList<string> GetComponentSchemas()
...
+            using JsonDocument document = JsonDocument.Parse(_content, _parseOptions);
+            JsonElement _root = document.RootElement;
+
+            if (_root.ValueKind == JsonValueKind.Array)
+            {
+                foreach (JsonElement element in _root.EnumerateArray())
+                {
+                    if (element.ValueKind == JsonValueKind.Object)
+                    {
+                        using MemoryStream stream = new MemoryStream();
```
Comment (montgomp) [inline resolvers/dotnet/Azure.DigitalTwins.Resolver/ModelQuery.cs:117]: Wouldn't
```C#
using (MemoryStream stream = new MemoryStream()) {
  ...
  using (StreamReader streamReader = new StreamReader(stream)) {
    ...
  }
}
```
free up these resources quicker by utilizing IDisposable on the `MemoryStream` and `StreamReader`?

Comment (montgomp) [reply resolvers/dotnet/Azure.DigitalTwins.Resolver/ModelQuery.cs:117]: NM, C# 8 features FTW!

**Review by montgomp — APPROVED** [review]

File: resolvers/dotnet/readme.md [hunk 2]
```diff
@@ -122,13 +122,32 @@ catch (ResolverException resolverEx)
...
+- `CalculateDependencies` [default: `true`] - Indicates desire to resolve all `DTMI`'s referenced in the to be resolved model document(s).
+- `UsePreComputedDependencies` [default: `false`] - Indicates the desire to use pre-calculated dependency payloads stored in the model repo **if they exist**.
+
+Here is an example using custom settings:
+
+```csharp
+using Azure.DigitalTwins.Resolver;
+
+ResolutionSettings customSettings =
+  new ResolutionSettings(usePreComputedDependencies: true, calculateDependencies: false);
```
Comment (rido-min) [inline resolvers/dotnet/readme.md:144]: Not clear what's the purpose of calculateDependencies to false. 

Consider using a single enum in the ctor to configure resolution. Try to fetch the `.expand.json` first or not. I'd enable it by default, and allow to opt-out with an options enum. Eg.

```
new ResolutionSettings(ResolutionSettingOption.DisableFetchWithExpand);
```

**Review by rido-min — COMMENTED** [review]
see https://github.com/Azure/iot-plugandplay-models-tools/issues/95 for feedback on CLI usage

I don't see any blocking issue for internal consumption.

**Pull request merged**